packages-ts/starknet-gauntlet/src/commands/base/executeCommand.ts

```diff
@@ -111,6 +113,12 @@ export const makeExecuteCommand = <UI, CI>(config: ExecuteCommandConfig<UI, CI>)
         contract: new Contract(c.contract.abi, c.contractAddress ?? '', c.provider.provider),
       }
 
+      const rdd = flags.rdd || process.env.RDD
```

Comment:
Imo it's not a good place to inject an RDD dependency here since it's a user input.

Reply:
as I understand it, this makes rdd available in the execution context which links back to the two different approaches to RDD handling - https://github.com/smartcontractkit/chainlink-starknet/pull/283#discussion_r1210360452

lets resolve in one thread

Reply:
I much prefer this solution since it injects RDD across **all** commands. It's a special input (`--rdd`) but most often actually set as an env var. It doesn't make sense as part of `makeUserInput` since we're not actually changing the input but we're instead sourcing the data from somewhere else. In my opinion entangling this into `makeUserInput` under each command resulted in a lot more code being added on Cosmos, and it was also hard to debug since your existing input would just disappear from an early return and other flags would be applied.

Reply:
The initial idea of `makeUserInput` was to resolve the conflicting inputs, do basic validation and raise exceptions if something is missing, then the input is being transformed into the actual contract input with `makeContractInput`. It's been consistently (well, more or less) done across our previous integrations and that it why I mention it here too. Anyways, I won't block you on that since this whole solution is about to be deprecated.